colibre/registration.py

```diff
@@ -118,3 +118,77 @@
     smhm.name = name
 
     setattr(self, f"stellar_mass_to_halo_mass_{aperture_size}_kpc", smhm)
+
+# if present iterate through available dust types
+try:
+    dust_fields = []
+    for sub_path in dir(catalogue.dust_mass_fractions):
+        if sub_path.startswith("dust_"):
+            dust_fields.append(getattr(catalogue.dust_mass_fractions, sub_path))
+    total_dust_fraction = sum(dust_fields)
+except AttributeError:
+    total_dust_fraction = np.zeros(stellar_mass.size)
+
+total_dust_mass = total_dust_fraction * catalogue.masses.m_star
+total_dust_mass.name = "$M_{\\rm dust}$ not found"
+
+
+setattr(self, f"total_dust_masses", total_dust_mass)
+
+# species fraction properties
+gas_mass = catalogue.masses.m_gas
+gal_area = (
+    2 * np.pi * catalogue.projected_apertures.projected_1_rhalfmass_star_100_kpc ** 2
+)
+mstar_100 = catalogue.projected_apertures.projected_1_mass_star_100_kpc
+
+self.xgass_galaxy_selection = np.logical_and(
+    catalogue.apertures.mass_star_100_kpc > unyt.unyt_quantity(10 ** 9, "Solar_Mass"),
+    catalogue.apertures.mass_star_100_kpc
+    < unyt.unyt_quantity(10 ** (11.5), "Solar_Mass"),
+)
+
+self.xcoldgass_galaxy_selection = np.logical_and(
+    catalogue.apertures.mass_star_100_kpc > unyt.unyt_quantity(10 ** 9, "Solar_Mass"),
+    catalogue.apertures.mass_star_100_kpc
+    < unyt.unyt_quantity(10 ** (11.5), "Solar_Mass"),
+)
```

Comment:
These still appear to be the same thing - and I have no idea what they are. Can we please have more descriptive names and a reason for why this must be repeated?